// win-shortcut/index.js
var exec = require('child_process').exec;


/*
{
    target_file: 'full path to target file',
    target_folder: 'full path to folder to place the shortcut',
    title: 'the name of the shortcut',
    icon: 'full path to icon file',
    icon_index: 1, // index of the icon file
    args: 'more args'
}
*/

var create = function(options, callback=null) {
    var query = '"./exe/nircmd.exe" shortcut ';
    if (!options.target_file || !options.target_folder) {
        if (callback) {
            return callback('Missing target file or target folder');
        }
    }

    query += '"' + options.target_file + '" "' + options.target_folder + '" ';

    if (options.title) {
        query += '"' + options.title + '"';
    } else {
        query += '"New Shortcut"';
    }

    if (options.args) {
        query += ' "' + options.args + '"';
    } else {
        query += ' ""';
    }

    if (options.icon) {
        query += ' "' + options.icon + '"';
        if (options.icon_index) {
            query += ' ' + options.icon_index;
        } else {
            query += ' 1';
        }
    }

    exec(query, (error, stdout, stderr) => {
        if (callback) {
            return callback(error ? stderr || stdout : null);
        }
    });
}

module.exports = {
    create: create
}
